WordPress/Classes/Models/WPAccount+Lookup.swift

```diff
@@ -0,0 +1,157 @@
+        if let displayName = defaultAccount.displayName {
+            defaultAccount.displayName = displayName.stringByDecodingXMLCharacters()
+        }
+
+        return defaultAccount
+    }
+
+    /// Lookup a WPAccount with the specified username, if it exists
+    ///
+    /// - Parameters:
+    ///   - username: The username associated with the account
+    ///   - context: An NSManagedObjectContext containing the `WPAccount` object with the given `username`.
+    /// - Returns: The `WPAccount` object associated with the given `username`, if it exists.
+    ///
+    static func lookup(withUsername username: String, in context: NSManagedObjectContext) throws -> WPAccount? {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.predicate = NSPredicate(format: "username = [c] %@ || email = [c] %@", username, username)
+        fetchRequest.includesPendingChanges = true // TODO: this shouldn't be necessary, it was copied from the `AccountService`
+
+        guard let account = try context.fetch(fetchRequest).first else {
+            return nil
+        }
+
+        return account
+    }
+
+    /// Lookup a WPAccount with the specified userID, if it exists
+    ///
+    /// - Parameters:
+    ///   - userID: The userID associated with the account
+    ///   - context: An NSManagedObjectContext containing the `WPAccount` object with the given `userID`.
+    /// - Returns: The `WPAccount` object associated with the given `userID`, if it exists.
+    ///
+    static func lookup(withUserID userID: Int64, in context: NSManagedObjectContext) throws -> WPAccount? {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.predicate = NSPredicate(format: "userID = %ld", userID)
+        fetchRequest.includesPendingChanges = true // TODO: this shouldn't be necessary, it was copied from the `AccountService`
+
+        guard let account = try context.fetch(fetchRequest).first else {
+            return nil
+        }
+
+        return account
+    }
+
+    /// Lookup the total number of `WPAccount` objects in the given `context`.
+    ///
+    /// If none exist, this method returns `0`.
+    ///
+    /// - Parameters:
+    ///   - context: An NSManagedObjectContext that may or may not contain `WPAccount` objects.
+    /// - Returns: The number of `WPAccount` objects in the given `context`.
+    ///
+    static func lookupNumberOfAccounts(in context: NSManagedObjectContext) throws -> Int {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.includesSubentities = false
+        return try context.count(for: fetchRequest)
+    }
+
+    // MARK: - Objective-C Compatibility Wrappers
+
+    /// An Objective-C wrapper around the `lookupDefaultWordPressComAccount` method.
+    ///
+    /// Prefer using `lookupDefaultWordPressComAccount` directly
```

Comment:
I haven't tried this, but I wonder if `@available` could be leveraged her to mark the objc wrappers as deprecated in swift but available to objc. I'm not sure of the behavior when called from objc code. 🤔

Reply:
We do this kind of thing in `UIColor+MurielColorsObjC.swift`:

```
@available(swift, obsoleted: 1.0)
```